openmrs-19/src/test/java/org/motechproject/openmrs19/tasks/OpenMRSTaskDataProviderTest.java

```diff
@@ -47,6 +52,57 @@ public void shouldReturnNullWhenClassIsNotSupported() {
     }
 
     @Test
+    public void shouldReturnNullWhenWrongLookupNameForEncounter() {
+        String className = OpenMRSEncounter.class.getSimpleName();
+        Object object = taskDataProvider.lookup(className, "wrongLookupName", null);
+
+        assertNull(object);
+        verifyZeroInteractions(encounterService);
+    }
+
+    @Test
+    public void shouldReturnNullWhenEmptyLookupFieldsForLookupGetEncounterByUuid() {
+        String className = OpenMRSEncounter.class.getSimpleName();
+        Map<String, String> lookupFields = new HashMap<>();
+
+        when(encounterService.getEncounterByUuid(null)).thenReturn(null);
+        Object object = taskDataProvider.lookup(className, BY_UUID, lookupFields);
+
+        assertNull(object);
+        verify(encounterService).getEncounterByUuid(null);
+    }
+
+    @Test
+    public void shouldReturnNullWhenEncounterNotFoundForLookupGetEncounterByUuid() {
+        String className = OpenMRSEncounter.class.getSimpleName();
+        Map<String, String> lookupFields = new HashMap<>();
+        lookupFields.put(UUID, "4");
+
+        when(encounterService.getEncounterByUuid("4")).thenReturn(null);
+        Object object = taskDataProvider.lookup(className, BY_UUID, lookupFields);
+
+        assertNull(object);
+        verify(encounterService).getEncounterByUuid("4");
+    }
+
+    @Test
+    public void shouldReturnEncounterForLookupGetEncounterByUuid() {
+        String className = OpenMRSEncounter.class.getSimpleName();
+        Map<String, String> lookupFields = new HashMap<>();
+        lookupFields.put(UUID, "5");
+
+        OpenMRSEncounter openMRSEncounter = new OpenMRSEncounter();
+        openMRSEncounter.setEncounterType("encounterTypeTest");
+
+        when(encounterService.getEncounterByUuid("5")).thenReturn(openMRSEncounter);
+        Object object = taskDataProvider.lookup(className, BY_UUID, lookupFields);
+
+        assertTrue(object instanceof OpenMRSEncounter);
```

Comment:
shouldn't we check that object equals to openMRSEncounter ? It should be the same instance